renderer/brave_content_renderer_client.cc

```diff
@@ -148,6 +150,13 @@ void BraveContentRendererClient::RenderFrameCreated(
         base::BindRepeating(&BraveRenderThreadObserver::GetDynamicParams));
   }
 
+  if (base::FeatureList::IsEnabled(
+          brave_rewards::features::kPlatformCreatorDetectionFeature) &&
+      !ChromeRenderThreadObserver::is_incognito_process()) {
+    new brave_rewards::RewardsRenderFrameObserver(
+        render_frame, content::ISOLATED_WORLD_ID_GLOBAL);
```

Comment:
One of the scripts (`twitter.ts`) must be run in the main world, since it accesses JS objects hanging off of DOM elements.

We'll need a security review I'm sure, but no new API is being added to the global environment. The scripts are compiled as a function body and then the browser capabilities are passed as arguments to the compiled function; they are only visible inside of the script.

Reply:
Updated to only use the main world for the one script.